src/Equinor.ProCoSys.IPO.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs

```diff
@@ -27,25 +28,30 @@ public class SavedFilterValidator : ISavedFilterValidator
         {
             var currentUserOid = _currentUserProvider.GetCurrentUserOid();
 
+            var project = await _context.QuerySet<Project>().SingleOrDefaultAsync(x => x.Name.Equals(projectName), cancellationToken);
+
             return await (from s in _context.QuerySet<SavedFilter>()
                 join p in _context.QuerySet<Person>() on EF.Property<int>(s, "PersonId") equals p.Id
                 where p.Oid == currentUserOid
                       && s.Title == title
-                      && s.ProjectName == projectName
+                      && s.ProjectId == project.Id
                 select s).AnyAsync(cancellationToken);
         }
         public async Task<bool> ExistsAnotherWithSameTitleForPersonInProjectAsync(int savedFilterId, string title,
             CancellationToken cancellationToken)
         {
             var currentUserOid = _currentUserProvider.GetCurrentUserOid();
-            var projectName = await (from s in _context.QuerySet<SavedFilter>() 
-                    where s.Id == savedFilterId select s.ProjectName).SingleOrDefaultAsync(cancellationToken);
+            var projectForSavedFilter = await (from s in _context.QuerySet<SavedFilter>()
```

Comment:
this change will cause nullpointer exception because of new line 54 when singleordefault return no projectForSavedFilter 
I think the correct solution is to return false if no projectForSavedFilter found. NB! Ensure there exists a unittest for that scenario in SavedFilterValidatorTests

Reply:
How about changing the filter in the next query instead?

Original: s.ProjectId == projectForSavedFilter.Id 
Suggested: projectForSavedFilter != null && s.ProjectId == projectForSavedFilter.Id

Reply:
Your choice. An unit-test covering the issue should give the same result both with my and your suggestion